spec/opentelemetry_spec.cr

```diff
@@ -53,7 +53,7 @@ describe OpenTelemetry do
     OpenTelemetry.trace "" do |span|
       (span["key"] ||= "value").should eq "value"
       (span["key"] ||= "another value").should eq "value"
-      span["key"].should eq "value"
+      span["key"]?.should eq "value"
```

Comment:
This was surprising to see in the diff. Was it needed?

Regardless, I do like that the error message will be better if it fails due to the key missing!